docs/rm/snax_cluster.md

```diff
@@ -0,0 +1,198 @@
+
+The RISCV CSR instruction has a 12-bit CSR address. There are other existing CSRs in Snitch, but we reserved CSR addresses `[0x3c0:0x5ff]` giving us a total of 575 different registers to use. Note that the Snitch does not instantiate any registers. Rather it offloads the CSR reads and writes unto the accelerator's.
+
+For example, the [HWPE MAC engine](https://github.com/KULeuven-MICAS/hwpe-mac-engine) has the following register mapping:
+
+| Register          | CSR Addresses      | Description                                                              |
+| ----------------- | -----------------  | -----------------------------------------------------------------------  |
+| Trigger           | 0x3c0              | Write 0 to start                                                         |
+| Acquire           | 0x3c1              | Lock the accelerator                                                     |
+| Finsihed          | 0x3c2              | Finished signal (1 pulse only)                                           |
+| Status            | 0x3c3              | 1: busy; 0: free                                                         |
+| Running           | 0x3c4              | 1: running; 0: free                                                      |
+| Softclear         | 0x3c5              | write 1 to reset the accelerator                                         |
+| Reserved          | 0x3c6              | Reserved                                                                 |
+| SWEVT             | 0x3c7              | Events                                                                   |
+| Generic           | 0x3c8 - 0x3cf      | General purpose registers (RW access)                                    |
+| Address A         | 0x3d0              | Starting address for input A                                             |
+| Address B         | 0x3d1              | Starting address for input B                                             |
+| Address C         | 0x3d2              | Starting address for input C                                             |
+| Address D         | 0x3d3              | Starting address for output D                                            |
+| Iterations        | 0x3d4              | Number of iterations                                                     |
+| Vector Length     | 0x3d5              | Number of elements                                                       |
+| Multiplier Mode   | 0x3d6              | 1: $d_i = a_i \cdot b_i$;  <br /> 0: $D = ( \sum a_i \cdot b_i) + C$     |
+| Vector Stide      | 0x3d7              | Vectore tride                                                            |
+
+There are no registers in the unused addresses `[0x3d8:0x5ff]`. However, a user can utilize the unused addresses for their needs.
```

Comment:
What if we have several accelerators in each Snitch core? Shall we need to split the address space for different accelerators?

Reply:
I agree, can we update this to be a relative offset @rgantonio ?

Reply:
OK! good idea I will put it.

Reply:
also @xiaoling-yi  this is true, maybe I'll clarify the definition for this.